collector/redis/integration_test.go

```diff
@@ -0,0 +1,89 @@
+package redis__test
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	tc "github.com/testcontainers/testcontainers-go/modules/compose"
+)
+
+type void struct{}
```

Comment:
To keep this particular syntax would be nice to have a comment about the meaning of the construct. Not really interested in notes on minimizing memory so much as the meaning... to me it's something along the lines of "let's us use map keys like a set."

To Dmitry's point, I don't know of an advantage (aside from keyboard ergonomics) to defining a type for the empty rather than typing the literal. 

I say ATM no need to change. We'll do some realtime restructuring changes and get things merged to a permanent home in just a little while. We can look at this again at that time.